src/sysusers/sysusers.c

```diff
@@ -370,341 +370,416 @@ static int rename_and_apply_smack(const char *temp_path, const char *dest_path)
         return r;
 }
 
-static int write_files(void) {
```

Comment:
could you extend the commit msg a bit? The patch is really hard to review, since the diff loses all structure of the patch, hence it would be good if you'd explain what precisely you are doing here, i.e. which parts you split out...